src/common.c

```diff
@@ -705,6 +705,20 @@ int fi_epoll_add(struct fi_epoll *ep, int fd, uint32_t events, void *context)
 	return 0;
 }
 
+int fi_epoll_mod(struct fi_epoll *ep, int fd, uint32_t events, void *context)
+{
+	int i;
+
+	for (i = 0; i < ep->nfds; i++) {
+		if (ep->fds[i].fd == fd) {
+			ep->fds[i].events = events;
+			ep->context[i] = context;
```

Comment:
This only works if we're not already waiting on the epoll.  Otherwise, the update won't apply until the wait call returns.

Reply:
For a generic solution, I think we would need to go this route.  If we're sure that modify won't be called while there's a waiting thread, we could document that limitation above fi_epoll_mod().  That would delay the need to provide a more robust implementation until it's needed.